# proteomicsliteratureminer
proteomicsliteratureminer is a new tool that aims to help researchers reduce time spent on literature research post analysis and streamline the decision about which proteins or genes are the most interesting and most promising for follow-up experiments.

<!-- badges: start -->
<!-- badges: end -->

The goal of proteomicsliteratureminer is to streamline the process of literature retrieval and provides result categorisation to assist researchers select appropraite leads for further research.

## Installation

You can install the released version of proteomicsliteratureminer from [SIH-GIT](https://github.com/Sydney-Informatics-Hub/proteomicsliteratureminer) with:

``` r
install.packages(devtools) # only if devtools is not installed
devtools::install_github("Sydney-Informatics-Hub/proteomicsliteratureminer")
```

## Example

Some ways of using the proteomicsliteratureminer package is shown below.
potentialmarker is a R dataframe that is part of the R package, for description of its contents, run the following R command
``` r
library(proteomicsliteratureminer)
?potentialmarker
```

Ex.1. Using the R data frame provided by the package, no output spreadsheet and plots specified, the object returned from pubMedMiner() is not assigned to any variable.
``` r
library(proteomicsliteratureminer)
result <- pubMedMiner(potentialmarker)
```
The result variable has 2 list elements - 
1. summary_results - Summarizes the query results 
2. pubmed_results - Summarizes the PubMed results based on the UniProt Identifiers and key words specified by the user

Ex.2. Using the R data frame provided by the package, with output spreadsheet specifed and no plots, the object returned from pubMedMiner() is not assigned to any variable.
``` r
library(proteomicsliteratureminer)
pubMedMiner(potentialmarker)
```

Ex.3. Using the R data frame provided by the package, with output spreadsheet specifed and no plots.
``` r
library(proteomicsliteratureminer)
pubMedMiner(potentialmarker, output.file = "potential_marker_pubmed_results.xlsx")
```
The output spread sheet with the PubMed output will be saved in the current working directory.

Ex.4. Using the R data frame provided by the package, with output spreadsheet specifed and no plots, the object returned from pubMedMiner() is not assigned to any variable.
``` r
library(proteomicsliteratureminer)
pubMedMiner(potentialmarker, output.file = "potential_marker_pubmed_results.xlsx", plots.dir = "plots")
```
The output spread sheet with the PubMed output will be saved in the current working directory. If an output spread sheet existed, it would be overwritten.
The images generated by the package will be saved in directory plots in the current working directory. If no plots directory was present, a new plots directory would be created.

Ex.5. Reading an Excel and converting it to a R dataframe
``` r
library(proteomicsliteratureminer)
library(openxlsx)
df <- readWorkbook(system.file("extdata", "input_uniprot_keywords.xlsx", package="proteomicsliteratureminer"))
result <- pubMedMiner(df, output.file = "input_uniprot_keywords_pubmed_results.xlsx", plots.dir = "plots")
```

## Citing
When using Lproteomicsliteratureminer please cite: Steffen P, Wu J, Hariharan S, Molloy MP, Schluter H, proteomicsliteratureminer A bioinformatics tool for prioritizing biological leads from omics data using literature mining.
